### PR TITLE
refactor: tefi-server tx api moved to tefi-electron-app

### PR DESCRIPTION
**Pull request by abrehmaaan**

### CODE REVIEW

File: transactions/fetchTx.ts [hunk 1]
```diff
@@ -14,12 +12,11 @@ export const formatTxData: (a: any) => txData = (txResult: any) => {
...
 };
 
-export const fetchTx = async (txHash: string, network = 'mainnet') => {
+export const fetchTx = async (txHash: string, _network = 'mainnet') => {
   try {
-    const url = network === 'mainnet' ? TX_FETCH_URL : TEST_TX_FETCH_URL;
-    const { data } = await axios.get(url + txHash);
-    const txData = formatTxData(data);
-    return txData;
+    const lcdUrl = process.env.NEXT_PUBLIC_SOMETHINGSSS ? TESTNET_LCD_URL : LCD_URL;
```
Comment (Fahad-Mahmood) [inline transactions/fetchTx.ts]: correct name is process.env.NEXT_PUBLIC_IS_TESTNET

**Review by Fahad-Mahmood — COMMENTED** [review]
I'm getting this error on the club screen.

Screenshot 2023-02-17 at 4 08 12 PM

File: pages/api/feed/posts.tsx [hunk 2]
```diff
@@ -33,17 +30,22 @@ const filterAndFormatPost = (data) => {
...
 
+const _getPost = async (offset = 0, limit = 100) => {
+  try {
+    const { data } = await axios.get(`${FCD_URL}v1/txs?offset=${offset}&limit=${limit}&account=${ADDRESS}`);
+    return { ...data };
+  } catch (err) {
+    return { err };
+  }
+};
+
```
Comment (Fahad-Mahmood) [inline pages/api/feed/posts.tsx]: remove try/catch from this fn

File: pages/api/feed/posts.tsx [hunk 3]
```diff
@@ -33,17 +30,22 @@ const filterAndFormatPost = (data) => {
...
-      headers: {
-        'X-Requested-With': 'XMLHttpRequest',
-      },
-    });
+    const result = await _getPost(offset, limit);
+    const postRequest = result ?? [];
     const next = postRequest?.data?.next ?? false;
     const posts = postRequest?.data && filterAndFormatPost(postRequest?.data);
-    return { posts, next };
+    return { posts: JSON.parse(JSON.stringify(posts)), next: JSON.parse(JSON.stringify(next)) };
```
Comment (Fahad-Mahmood) [inline pages/api/feed/posts.tsx]: we don't need this here, ```return {posts, next};```

File: pages/api/feed/posts.tsx [hunk 4]
```diff
@@ -33,17 +30,22 @@ const filterAndFormatPost = (data) => {
...
 export const getPost = async (offset = 0, limit = 100) => {
   try {
-    const query = `${CLUB_SERVER_ROOT}/posts?offset=${offset}&limit=${limit}`;
-    const postRequest = await axios.get(query, {
-      headers: {
-        'X-Requested-With': 'XMLHttpRequest',
-      },
-    });
+    const result = await _getPost(offset, limit);
+    const postRequest = result ?? [];
```
Comment (Fahad-Mahmood) [inline pages/api/feed/posts.tsx]: We don't want to create another api, please create a hook ```usePosts``` in the data folder and move the getPost function there.

File: pages/api/feed/posts.tsx [hunk 5]
```diff
@@ -1,61 +1,58 @@
...
+//   if (
+//     post?.tx?.value?.memo &&
+//     post?.tx?.value?.msg[0].type == 'bank/MsgSend' &&
+//     post?.tx?.value?.msg[0]?.value.amount[0].denom == 'uusd' &&
+//     post?.tx?.value?.msg[0]?.value.amount[0].amount / MICRO >= parseFloat(FILTER_POST_UST)
+//   ) {
+//     return true;
+//   }
+//   return false;
+// };
```
Comment (Fahad-Mahmood) [inline pages/api/feed/posts.tsx]: this file can be deleted

**Review by Fahad-Mahmood — APPROVED** [review]
Please remove the comments before merging the PR. We will test this again with the working api.

**Comment by abrehmaaan** [comment]
> Please remove the comments before merging the PR. We will test this again with the working api.

Okay @Fahad-Mahmood

**Pull request merged**